### ISSUE TITLE
Fix systemUiVisibility deprecations in ThemeManager

### ISSUE THREAD

**Issue by Amejia481** [mozilla-mobile/fenix#17808]
Deprecation docs:
https://developer.android.com/reference/android/view/View#getSystemUiVisibility()
https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LIGHT_STATUS_BAR

https://github.com/mozilla-mobile/fenix/blob/f4a7b4c066b98e8fd999268d702f4d5d81815021/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt#L81

https://github.com/mozilla-mobile/fenix/blob/f4a7b4c066b98e8fd999268d702f4d5d81815021/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt#L88

https://github.com/mozilla-mobile/fenix/blob/f4a7b4c066b98e8fd999268d702f4d5d81815021/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt#L96

https://github.com/mozilla-mobile/fenix/blob/f4a7b4c066b98e8fd999268d702f4d5d81815021/app/src/main/java/org/mozilla/fenix/theme/ThemeManager.kt#L102

Similar https://github.com/mozilla-mobile/android-components/issues/9554

**Comment by CharlesAdam** [comment]
Can you assign me to this issue?

**Comment by Amejia481** [comment]
Sure!

**Comment by CharlesAdam** [comment]
I'm trying to update the compileSdkVersion so I can use the new features to solve this issue since they are only available at API 30 however some errors are appearing whenever the projects try to build as in this screenshot:
![image](https://user-images.githubusercontent.com/43017375/108234461-94d96000-7123-11eb-8fc1-63f5fd1d3f54.png)

**Comment by CharlesAdam** [comment]
@Amejia481 It seems some methods are not being overridden because now activities cannot be null as in this link: https://developer.android.com/reference/android/app/Application.ActivityLifecycleCallbacks#onActivityCreated(android.app.Activity,%20android.os.Bundle)
So we just need to remove the safe calls so the app can be built, however, should this be another issue?

**Comment by cadeyrn** [comment]
> should this be another issue?

There is already a PR: #17809

**Comment by CharlesAdam** [comment]
@cadeyrn thanks for the info, I will generate the branch from this PR :smiley:

**Comment by Amejia481** [comment]
Thank you @cadeyrn and CharlesAdam, we are doing some round of tests to find any possible issue before land it.

**Comment by mcarare** [comment]
@CharlesAdam are you still working on this?

**Comment by CharlesAdam** [comment]
@mcarare Yes, I sent a PR back then but it shouldn't be merged before the base branch was merged, I'll update my branch and set it up for the merge

**Comment by mcarare** [comment]
This is still waiting on a stable version for https://developer.android.com/jetpack/androidx/releases/core 1.5.0 (it has reached the RC stage).

**Comment by wbillan** [comment]
1.5.0 stable was released May 18

**Comment by mcarare** [comment]
@CharlesAdam Do you want to continue working on this?

**Comment by mcarare** [comment]
Closed in https://github.com/mozilla-mobile/fenix/pull/20220.
